lib/ovirt_provider/events/ovirt_event_monitor.rb

```diff
@@ -0,0 +1,37 @@
+class OvirtEventMonitor
+  def initialize(options={})
+    @options = options
+  end
+
+  def inventory
+    @inventory ||= Ovirt::Inventory.new(@options)
+  end
+
+  def start
+    trap(:TERM) { $rhevm_log.info "EventMonitor#start: ignoring SIGTERM" }
+    @since          = nil
+    @inventory      = nil
+    @monitor_events = true
+  end
+
+  def stop
+    @monitor_events = false
+  end
+
+  def each_batch
+    while @monitor_events do
+      # grab only the most recent event if this is the first time through
+      query_options = @since ? {:since => @since} : {:max => 1}
```

Comment:
I may do that in a separate PR.  The original behavior was the same (would miss similar gaps).